go/vt/vttablet/tabletserver/vstreamer/rowstreamer.go

```diff
@@ -204,95 +204,105 @@ func (rs *rowStreamer) buildSelect() (string, error) {
 }
 
 func (rs *rowStreamer) streamQuery(conn *snapshotConn, send func(*binlogdatapb.VStreamRowsResponse) error) error {
-	log.Infof("Streaming query: %v\n", rs.sendQuery)
-	gtid, err := conn.streamWithSnapshot(rs.ctx, rs.plan.Table.Name, rs.sendQuery)
-	if err != nil {
-		return err
-	}
-
-	// first call the callback with the fields
-	flds, err := conn.Fields()
-	if err != nil {
-		return err
-	}
-	pkfields := make([]*querypb.Field, len(rs.pkColumns))
-	for i, pk := range rs.pkColumns {
-		pkfields[i] = &querypb.Field{
-			Name: flds[pk].Name,
-			Type: flds[pk].Type,
-		}
-	}
-
-	err = send(&binlogdatapb.VStreamRowsResponse{
-		Fields:   rs.plan.fields(),
-		Pkfields: pkfields,
-		Gtid:     gtid,
-	})
-	if err != nil {
-		return fmt.Errorf("stream send error: %v", err)
-	}
 
-	response := &binlogdatapb.VStreamRowsResponse{}
-	lastpk := make([]sqltypes.Value, len(rs.pkColumns))
-	byteCount := 0
+	offset := 0
+	limit := 10000000
+	delta := 10000001
```

Comment:
i think this ends up skipping a row due to the offset ending up being `liimit + 1`. consider the following example where we try to visit all rows in a set of 10 rows in a test table:

```
mysql> show create table bramos_repl_table;
+-------------------+----------------+
| Table             | Create Table   |
|-------------------+----------------|
| bramos_repl_table | CREATE TABLE `bramos_repl_table` (
  `date_latest` int(11) DEFAULT NULL,
  `foo` int(11) DEFAULT '0',
  `date_first` int(11) NOT NULL DEFAULT '0',
  `test_int` int(11) DEFAULT '0',
  `test_string` varchar(20) DEFAULT '',
  KEY `date_latest` (`date_latest`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4                |
+-------------------+----------------+
1 row in set
Time: 0.003s
mysql> select date_latest from bramos_repl_table  order by date_latest limit 10;
+---------------+
|   date_latest |
|---------------|
|    1621546586 |
|    1621546587 |
|    1621546588 |
|    1621546589 |
|    1621546590 |
|    1621546591 |
|    1621546592 |
|    1621546593 |
|    1621546594 |
|    1621546595 |
+---------------+
10 rows in set
Time: 0.004s
mysql> select date_latest from bramos_repl_table  order by date_latest limit 5 offset 0;
+---------------+
|   date_latest |
|---------------|
|    1621546586 |
|    1621546587 |
|    1621546588 |
|    1621546589 |
|    1621546590 |
+---------------+
5 rows in set
Time: 0.003s
mysql> select date_latest from bramos_repl_table  order by date_latest limit 5 offset 5;
+---------------+
|   date_latest |
|---------------|
|    1621546591 |
|    1621546592 |
|    1621546593 |
|    1621546594 |
|    1621546595 |
+---------------+
5 rows in set
Time: 0.004s
mysql> select date_latest from bramos_repl_table  order by date_latest limit 5 offset 6;
+---------------+
|   date_latest |
|---------------|
|    1621546592 |
|    1621546593 |
|    1621546594 |
|    1621546595 |
|    1621546596 |
+---------------+
5 rows in set
Time: 0.003s
```